base/linalg/rowvector.jl

```diff
@@ -147,6 +154,11 @@ end
 
 # Multiplication #
 
+# inner product -> dot product specializations
+@inline *{T<:Real}(rowvec::RowVector{T}, vec::AbstractVector{T}) = dot(parent(rowvec), vec)
+@inline *(rowvec::ConjRowVector, vec::AbstractVector) = dot(rowvec', vec)
```

Comment:
I'm worried about potential performance regressions if we don't also have `*(rowvec,Matrix) = At_mul_B` and `*(conjrowvec,Matrix) = Ac_mul_B`